protocols/gossipsub/src/lib.rs

```diff
@@ -122,6 +122,10 @@
 //! println!("Listening on {:?}", addr);
 //! ```
 
+// Re-export open_metrics_client so external applications can inject the correct Registry when
+// using metrics
+pub use open_metrics_client;
```

Comment:
I would guess users that use `libp2p-gossipsub` metrics also use `libp2p-metrics`? In addition I would assume that users consume both crates via the root level `libp2p` which combines versions of `libp2p-gossipsub` and `libp2p-metrics` using the same `open-metrics-client` version. With that in mind should this re-export not happen within `libp2p-metrics`?

Reply:
Because we had to inject the `Regsitry` directly into gossipsub, the `libp2p-metrics` addition is rather superfluous (if you're using the metrics inside gossipsub). As it stands, `libp2p-metrics` only logs events seen by the behaviour (i.e message count) and these are relatively redundant compared to the detailed metrics we can get inside gossipsub now. 
I was going to remove the gossipsub addition to `libp2p-metrics` but figured it could still be useful for others who just want a high-level overview of their gossipsub implementation. 

Anyway, we are currently consuming this directly by the `libp2p-gossipsub` crate which is why I re-exported it there. Perhaps there is a better way to handle this? I can also add the re-export to `libp2p-metrics` if you think appropriate.

Reply:
I don't have a strong opinion on re-exports. If I am not mistaken, unless libp2p specific, I think we don't do them at all today.

Off the top of my head, I would prefer not re-exporting `open-metrics-client` at all. Users that try to use a different version than the one used in `libp2p-gossipsub` will get a compile time error like with any other version mismatch. Do we need to special case this here?

Reply:
Yep sure. Removed the re-export. 

I guess I was being lazy, where I tie the dependency solely to libp2p and upgrading the libp2p dep upgrades the required open metric dep for me. I can manage the dep version in my project too.